src/core/dbus-execute.c

```diff
@@ -977,9 +1051,13 @@ int bus_set_transient_exec_command(
                         if (!a)
                                 return -ENOMEM;
 
-                        fprintf(f, "%s=%s@%s %s\n",
+                        fprintf(f, "%s=%s%s%s%s%s@%s %s\n",
                                 name,
-                                c->flags & EXEC_COMMAND_IGNORE_FAILURE ? "-" : "",
+                                FLAGS_SET(c->flags, EXEC_COMMAND_IGNORE_FAILURE)   ? "-" : "",
+                                FLAGS_SET(c->flags, EXEC_COMMAND_NO_ENV_EXPAND)    ? ":" : "",
+                                FLAGS_SET(c->flags, EXEC_COMMAND_FULLY_PRIVILEGED) ? "+" : "",
+                                FLAGS_SET(c->flags, EXEC_COMMAND_NO_SETUID)        ? "!" : "",
+                                FLAGS_SET(c->flags, EXEC_COMMAND_AMBIENT_MAGIC)    ? "!!" : "",
```

Comment:
maybe introduce a helper call here: `char *exec_command_flags_to_exec_chars(ExecCommandFlags flags)` or so, which concats the flags like this and returns this in an allocated buffer